crates/rspack_core/src/compiler/mod.rs

```diff
@@ -343,36 +344,37 @@ impl Compiler {
       .await?;
 
     let mut new_emitted_asset_versions = HashMap::default();
-    let results = self
-      .compilation
-      .assets()
-      .iter()
-      .filter_map(|(filename, asset)| {
-        // collect version info to new_emitted_asset_versions
-        if self
-          .options
-          .experiments
-          .incremental
-          .contains(IncrementalPasses::EMIT_ASSETS)
-        {
-          new_emitted_asset_versions.insert(filename.to_string(), asset.info.version.clone());
-        }
+    // SAFETY: await immediately, trust caller to handle the future correctly
+    let _ = unsafe {
+      TokioScope::scope_and_collect(|s| {
```

Comment:
Are we supposed to switch over to our custom scoped future impl?

Reply:
yeah and async-scoped doesn't support wasm 😭